props/saber_sabersense_buttons.h

```diff
@@ -349,47 +351,66 @@ GESTURE CONTROLS
   // Check user-defined array is valid at compile time.
   static_assert(
     SABERSENSE_DEFAULT_BLADE_ARRAY < NELEM(blades),
-    "[Sabersense] ERROR: "
-    "#define SABERSENSE_DEFAULT_BLADE_ARRAY must be less than the number of blade arrays present."
+    "SABERSENSE_DEFAULT_BLADE_ARRAY must reference a valid array (note zero-based counting)."
   );
+
 #ifdef BLADE_DETECT_PIN
   static_assert(
     SABERSENSE_DEFAULT_BLADE_ARRAY != 0,
-    "[Sabersense] ERROR: "
-    "#define SABERSENSE_DEFAULT_BLADE_ARRAY must be 1 or higher when using Blade Detect."
+    "[Sabersense] ERROR: Default array index must be 1 or higher when using Blade Detect."
   );
+  constexpr int min_blades_required = 3;
+#else
+  constexpr int min_blades_required = 2;
 #endif
 
+  // Check that Array Selector is actually needed (prevents possible unexpected behaviour).
+  static_assert(
+    NELEM(blades) >= min_blades_required,
+    "[Sabersense] ERROR: Array Selector is not required with only one selectable blade array. "
+    "Please undefine SABERSENSE_ARRAY_SELECTOR."
+  );
+
 #ifndef SABERSENSE_DISABLE_SAVE_ARRAY
 class SaveArrayStateFile : public ConfigFile {
 public:
   void iterateVariables(VariableOP *op) override {
-    // Default array if no save file present...
     CONFIG_VARIABLE2(sabersense_array_index, SABERSENSE_DEFAULT_BLADE_ARRAY);
   }
-  int sabersense_array_index;  // Stores current array index.
+  int sabersense_array_index;
 };
 #endif
 
-    struct SabersenseArraySelector {
-      static int return_value;  // Tracks current array index.
-      float id() {
-        if (return_value < 0 || return_value >= NELEM(blades)) {
-          Serial.println("[Sabersense] ALERT: User or externally-specified array index invalid. "
-                         "Resetting to default.");
-          return_value = SABERSENSE_DEFAULT_BLADE_ARRAY;
-        }
-        return return_value;
+  struct SabersenseArraySelector {
+    static int return_value;
+    float id() {
+      if (return_value < 0 || return_value >= NELEM(blades)) {
+        Serial.println("[Sabersense] ALERT: User or externally-specified array index invalid. "
+                       "Resetting to default.");
+        return_value = SABERSENSE_DEFAULT_BLADE_ARRAY;
       }
-      static void cycle() {
+      return return_value;
+    }
+
+    static void cycle_directional(bool forward) {
 #ifndef BLADE_DETECT_PIN
-        return_value = (return_value + 1) % NELEM(blades);
-#else   // Improved logic fixes early Array1 repetition when using Blade Detect.
-        return_value = 1 + return_value % (NELEM(blades) - 1);
-#endif
+      const int size = NELEM(blades);
+      return_value = (return_value + (forward ? 1 : size - 1)) % size;
+#else
```

Comment:
The if and else do very similar things here. It seems like they should be able to share more code.
Maybe something like this:
```cpp
   int size = NELEM(blades);
   int offset = 0;
#ifdef BLADE_DETECT_PIN
   offset = 1;
   size--;
#endif
   return_value += forward ? 1 : (size - 1);
   return_value = (return_value + size - offset) % size + offset;
```